lib/ocpp/v16/database_handler.cpp

```diff
@@ -377,21 +398,31 @@ void DatabaseHandler::insert_or_update_local_authorization_list_entry(const CiSt
     }
 
     if (stmt->step() != SQLITE_DONE) {
-        EVLOG_error << "Could not insert into table: " << this->database->get_error_message();
-        throw std::runtime_error("db access error");
+        EVLOG_error << "Could not insert or update local authorization list entry into the database";
+        throw QueryExecutionException(this->database->get_error_message());
     }
 }
 
 void DatabaseHandler::insert_or_update_local_authorization_list(
     std::vector<v16::LocalAuthorizationList> local_authorization_list) {
+    bool success = true; // indicates if all database operations succeeded
     for (const auto& authorization_data : local_authorization_list) {
-        if (authorization_data.idTagInfo) {
-            this->insert_or_update_local_authorization_list_entry(authorization_data.idTag,
-                                                                  authorization_data.idTagInfo.value());
-        } else {
-            this->delete_local_authorization_list_entry(authorization_data.idTag.get().c_str());
+        try {
+            if (authorization_data.idTagInfo) {
+                this->insert_or_update_local_authorization_list_entry(authorization_data.idTag,
+                                                                      authorization_data.idTagInfo.value());
+            } else {
+                this->delete_local_authorization_list_entry(authorization_data.idTag.get().c_str());
+            }
+        } catch (const QueryExecutionException& e) {
+            // catch but continue with remaining entries
+            success = false;
         }
     }
+
+    if (!success) {
+        throw QueryExecutionException(this->database->get_error_message());
```

Comment:
Does this really always provide a meaningful error message? I'm not 100% sure here, for example if the first insert_or_update fails and all subsequent once succeed get_error_message (which uses sqlite3_errmsg) may have already been cleared: https://www.sqlite.org/capi3ref.html#sqlite3_errcode